### DISCUSSION TITLE
Feature requests to GHE repository caches.

### DISCUSSION THREAD

**Discussion by mayurva** [community/community#36276] [Enterprise]
Hello,

My name is Mayur, and I'm Engineering manager for the productivity engineering team at Box. My team manages the Github Enterprise server for Box and we have been incredibly excited to try out the repository caches for our appliance. While exploring the repository caches, we came up with the below listed feature requests for future which would greatly improve the usability and use cases for repository caches:

- The repository caches continue serving traffic independently of whether the primary is up or not.
- Similar to backup/restore utils, the repository caches have a backwards compatibility with 2 feature releases of GHE, so that the primary and cache upgrades can be scheduled independently.
- The maximum limit of 8 replicas is increased or removed altogether.

If these features are implemented, it would allow us to serve read-only copy of certain critical repositories across multiple regions even when GHE primary is down for maintenance.

While there are other solutions to distribute highly available read-only cache of git repositories, a native support in GHE would be a great feature addition.

**Comment by hosensahadat7** [comment]
Native support in GHE would be a great feature addition.

**Comment by ossanna16** [comment]
Hey there @mayurva and thank you for the suggestion! I'm glad you found our community 🙂
